sky/clouds/gcp.py

```diff
@@ -426,6 +426,85 @@ def _get_image_size(cls, image_id: str) -> float:
                             image_id=image_id)) from None
             raise
 
+    @classmethod
+    def check_volume_name(cls, region: 'clouds.Region',
+                          zones: Optional[List['clouds.Zone']], use_mig: bool,
+                          volume_name: str) -> Dict[str, Any]:
+        """Check if the volume name exists and return the volume info."""
+        logger.debug(
+            f'Checking volume {volume_name} in region {region} and zones {zones}'
+        )
+        try:
+            compute = gcp.build('compute',
+                                'v1',
+                                credentials=None,
+                                cache_discovery=False)
+        except gcp.credential_error_exception():
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError('Not able to build compute client') from None
+
+        project_id = cls.get_project_id()
+        # If zones is None, check the region disk
+        # If zones is not None, check zone disks first, then region disk
+        if zones is None:
+            # Set zones to an empty list to normalize the logic below
+            zones = []
```

Comment:
We should probably ban the case when the region is not specified, when a volume is specified in the file_mounts. Reason: we should avoid the case when a volume is accidentally created when people want to use an existing volume.

Reply:
I have updated code to raise error if neither region nor zone is specified for an existing volume.